b2share/modules/deposit/api.py

```diff
@@ -112,6 +128,36 @@ def create(cls, data, id_=None):
         # Set record's schema
         if '$schema' in data:
             raise InvalidDepositError('"$schema" field should not be set.')
+
+        if '_parent_pids' in data:
+            raise InvalidDepositError(
+                'Field "_parent_pids" cannot be set.')
+        version_master, prev_version = None, None
+        if version_of:
+            version_master, prev_version = find_version_master_and_previous_record(version_of)
+
+        if prev_version:
+            if not get_version_parent_pid(prev_version):
+                # previous version needs updating with the versioning parent pid
+                set_version_parent_pid(prev_version, version_master.parent)
+                prev_version.commit()
+                RecordIndexer().index(prev_version)
+
+            if 'community' in data and data['community'] != prev_version.model.json['community']:
+                raise InvalidDepositError(
+                    'Community ID must be identical with '
+                    'the previous version\'s community ID.')
+            if current_user.id not in prev_version['_deposit']['owners']:
+                raise OwnershipVersioningError()
+            patch = data
+            data = copy.deepcopy(prev_version.model.json)
+            # eliminate _deposit, _files, _oai, _pid, etc.
+            data = {k: v for k, v in data.items() if not k.startswith('_')}
+            data.update(patch)
```

Comment:
Do we really want to always make a patch. This might create issue with records created via the REST API.